dataframe/dataframe.go

```diff
@@ -1695,7 +1695,7 @@ func (df DataFrame) createRowKey(keys []string, keyIdx []int, rowIdx int) string
 
 func createValueKey(val series.Element) string {
 	if val.Type() == series.Float {
-		return fmt.Sprintf("%g", val.Float())
+		return strconv.FormatFloat(val.Float(), 'f', -1, 64)
 	}
 	return val.String()
```

Comment:
Why are we not using `val.String()` for float?

Reply:
Also, how does it handled `NaN` ?

Reply:
-> Why are we not using val.String() for float?
Because it will still contain trailing zero.

-> Also, how does it handled NaN ?
the output of createValueKey will be `NaN` also

---

```
func createValueKey(val series.Element) string {
	if val.Type() == series.Float {
		log.Println(val.String())
		log.Println(fmt.Sprintf("%g", val.Float()))
		return strconv.FormatFloat(val.Float(), 'f', -1, 64)
	}
	return val.String()
}
```

```
func Test_createValueKey(t *testing.T) {
	type args struct {
		val series.Element
	}
	tests := []struct {
		name string
		args args
		want string
	}{
		{
			"float: 1",
			args{series.Floats(1).Elem(0)},
			"1",
		},
		{
			"float: 1.2",
			args{series.Floats(1.2).Elem(0)},
			"1.2",
		},
		{
			"float: 900000000",
			args{series.Floats(900000000).Elem(0)},
			"900000000",
		},
		{
			"NaN",
			args{series.Floats("NaN").Elem(0)},
			"NaN",
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := createValueKey(tt.args.val); got != tt.want {
				t.Errorf("createValueKey() = %v, want %v", got, tt.want)
			}
		})
	}
}
```

Output:

```
Running tool: /usr/local/go/bin/go test -timeout 30s -run ^Test_createValueKey$ github.com/go-gota/gota/dataframe -v

=== RUN   Test_createValueKey
=== RUN   Test_createValueKey/float:_1
2023/02/21 12:31:57 1.000000
2023/02/21 12:31:57 1
=== RUN   Test_createValueKey/float:_1.2
2023/02/21 12:31:57 1.200000
2023/02/21 12:31:57 1.2
=== RUN   Test_createValueKey/float:_900000000
2023/02/21 12:31:57 900000000.000000
2023/02/21 12:31:57 9e+08
=== RUN   Test_createValueKey/NaN
2023/02/21 12:31:57 NaN
2023/02/21 12:31:57 NaN
--- PASS: Test_createValueKey (0.00s)
    --- PASS: Test_createValueKey/float:_1 (0.00s)
    --- PASS: Test_createValueKey/float:_1.2 (0.00s)
    --- PASS: Test_createValueKey/float:_900000000 (0.00s)
    --- PASS: Test_createValueKey/NaN (0.00s)
PASS
ok  	github.com/go-gota/gota/dataframe	0.482s
```